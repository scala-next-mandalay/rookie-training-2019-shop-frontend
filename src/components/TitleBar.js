import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Box, IconButton, AppBar, Toolbar, Badge,Divider,Button } from '@material-ui/core';
import { Menu as MenuIcon, ShoppingCart as ShoppingCartIcon } from '@material-ui/icons';
import { FormattedMessage } from 'react-intl';
import CartItemPropTypes from './CartItemPropTypes';
import IconLinkButton from './IconLinkButton';
import './style.css';
import { Link as RouterLink } from 'react-router-dom';
import NavTesting from '../containers/NavTesting';
import CheckoutOrder from '../containers/CheckoutOrder';
import Drawer from "@material-ui/core/Drawer";
import CloseIcon from '@material-ui/icons/Close';

const useStyles = makeStyles(theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
     backgroundColor:'#fff',
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('lg')]: {
      display: 'none',
    },
  },
  rightIcon: {
    marginLeft: 'auto',
    marginRight: theme.spacing(1),
  },
  list: {
    width: 300,
    [theme.breakpoints.up('sm')]: {
    width :600
    },
  },
  closebtn:{
   marginLeft: theme.spacing(2),
    flex: 1,
  }
}));

const TitleBar = ({cart, totalQuantity, handleDrawerToggle, showMenu ,showNav,showIcon}) => {
  const classes = useStyles();
  const [state, setState] = React.useState({
    right: false
  });

  const toggleDrawer = (side, open) => event => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    ) {
      return;
    }

    setState({ ...state, [side]: open });
  };

  const sideList = side => (
    <div className={classes.list}>
       <Box display="flex">
       <Box ml={1} mr="auto" my="auto"> 
       <IconButton edge="start"  onClick={toggleDrawer(side, false)} aria-label="Close" className={classes.closebtn}>
            <CloseIcon />
       </IconButton>
       </Box>
       <Box mr={2} ml="auto" my="auto">Your Cart List </Box></Box> 
            <Divider />
        <CheckoutOrder/>
        
      
        <Button
           onClick={toggleDrawer(side, false)}
            className="btn_Purchase"
            variant="contained" 
            color="primary"
            to="/checkout"
            fullWidth
            component={RouterLink}>
              SECURE CHECKOUT
            </Button>
         <Divider/>
    </div>
  );
  
  return (
    <div>
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        <IconButton
         
          aria-label="Open drawer"
          edge="start"
          onClick={handleDrawerToggle}
          //disabled={(handleDrawerToggle === null)}
          className={classes.menuButton}
        >
          {showMenu ? <MenuIcon />:null}
        </IconButton>

        <Box fontSize={{xs:"subtitle1", sm:"h6.fontSize"}} ml={0} display="flex" flexDirection="row" className="title">
          <FormattedMessage id="Top.Title" />
        </Box>
         <Box>
         {showNav ? <NavTesting /> : null}
        </Box>
          {showIcon ?
        <Box mr={0} ml="auto">
          <IconLinkButton color="primary" onClick={toggleDrawer("right", true)}>
            <Badge 
              color="secondary" 
              badgeContent={totalQuantity} 
              invisible={cart.length > 0 ? false: true} 
            >
              <ShoppingCartIcon />
            </Badge>
          </IconLinkButton>
        </Box>: null}
      </Toolbar>
    </AppBar>
    <Drawer
        anchor="right"
        open={state.right}
        onClose={toggleDrawer("right", false)}
      >
        {sideList("right")}
      </Drawer>
    </div>
  );
};
TitleBar.propTypes = {
  cart: PropTypes.arrayOf(CartItemPropTypes.isRequired).isRequired,
  totalQuantity: PropTypes.number.isRequired,
  showNav: PropTypes.bool.isRequired,
  showMenu: PropTypes.bool.isRequired,
   showIcon: PropTypes.bool.isRequired,
};

export default TitleBar;