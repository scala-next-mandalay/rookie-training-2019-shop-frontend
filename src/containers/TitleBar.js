import { connect } from 'react-redux'
import TitleBar from '../components/TitleBar'

export default connect(
  (state) => ({
    cart: state.cart.rows,
    totalQuantity:  state.cart.totalQuantity,
     totalPrice: state.cart.totalPrice
  }),
  null
)(TitleBar)
