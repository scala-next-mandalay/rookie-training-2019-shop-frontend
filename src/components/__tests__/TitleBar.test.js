import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import TitleBar from '../TitleBar';
/*global expect*/
/*global jest*/
/*global MouseEvent*/
/*global KeyboardEvent*/
let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
})


const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
const showIcon = true;
const showNav = true;
const showMenu = true;
  let handleDrawerToggle=jest.fn(()=>{
  
});
  const fetchAuthedUser=jest.fn(()=>{
  let user=[{id:1,email:"a@gmail.com"}]
})
  let signIn=jest.fn(()=>{
  let auth=true;
})
  let signOut=jest.fn(()=>{
  let auth=false;
})
const setAnchorEl=jest.fn(()=>{
  let anchorEl=null;
})
  const cart=[{ id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 },
             { id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }]

describe("Titlebar component", () => {
  
  it('matches the snapshot', () => {
    const TitlebarSnapshot= renderer.create(
      <Parent>
        <TitleBar cart={cart} 
        totalQuantity={2} showMenu={showMenu} showNav={showNav}
        showIcon={showIcon} fetchAuthedUser={fetchAuthedUser}
        handleDrawerToggle={handleDrawerToggle}/>
      </Parent>
    ).toJSON();
    expect(TitlebarSnapshot).toMatchSnapshot();

  });
  

});

  describe("Testing TitalBar",()=>{
  it('category list and shopping icon',()=>{
    act(()=>{
      ReactDOM.render((
        <Parent>
         <TitleBar cart={cart} 
          totalQuantity={2} showMenu={showMenu} showNav={showNav}
         showIcon={showIcon} fetchAuthedUser={fetchAuthedUser}
         handleDrawerToggle={handleDrawerToggle}
         setAnchorEl={setAnchorEl}/>
        </Parent>
        ),container);
    });
   
    const buttonArr = container.querySelectorAll('button');
    const span = container.querySelectorAll('span');
    expect(span[2].textContent).toBe("Rookie Training 2019");
     act(()=>{
      buttonArr[0].dispatchEvent(new MouseEvent('click',{bubbles:true}));
    });
    expect(handleDrawerToggle).toHaveBeenCalledTimes(1);
    
    const evt = new KeyboardEvent('keydown', { keyCode: 27 });
    // 27 == Escape Key    
    document.dispatchEvent(evt);
    expect(handleDrawerToggle).toHaveBeenCalledTimes(1);
    act(()=>{
      buttonArr[1].dispatchEvent(new MouseEvent('click',{bubbles:true}));
    });
    act(()=>{
      buttonArr[2].dispatchEvent(new MouseEvent('click',{bubbles:true}));
    });
    
 });
    
})  
