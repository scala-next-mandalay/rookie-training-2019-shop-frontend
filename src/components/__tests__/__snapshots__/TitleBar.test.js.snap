// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titlebar component matches the snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="Open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <div
        className="PrivateHiddenCss-smDown-76"
      >
        <div
          className="MuiBox-root MuiBox-root-87 title"
        >
          <span>
            Rookie Training 2019
          </span>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-88"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiBox-root MuiBox-root-192 makeStyles-navMenuList-6"
          >
            <div
              className="MuiBox-root MuiBox-root-193"
            >
              <div
                className="PrivateHiddenCss-onlySm-74 PrivateHiddenCss-onlyXs-71 PrivateHiddenCss-onlyMd-77"
              >
                <div
                  className="MuiGrid-root MuiGrid-container"
                >
                  <div
                    align="center"
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      className="root"
                    >
                      <div
                        className="MuiFormControl-root MuiTextField-root makeStyles-margin-195 makeStyles-textField-196 MuiFormControl-marginDense"
                      >
                        <div
                          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                          onClick={[Function]}
                        >
                          <select
                            aria-invalid={false}
                            className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense MuiSelect-outlined"
                            disabled={false}
                            multiple={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                          />
                          <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden={true}
                            className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                            style={
                              Object {
                                "paddingLeft": 8,
                              }
                            }
                          >
                            <legend
                              className="PrivateNotchedOutline-legend-243"
                              style={
                                Object {
                                  "width": 0.01,
                                }
                              }
                            >
                              <span
                                dangerouslySetInnerHTML={
                                  Object {
                                    "__html": "&#8203;",
                                  }
                                }
                              />
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  className="MuiBadge-badge MuiBadge-colorSecondary"
                >
                  2
                </span>
              </span>
            </span>
          </button>
          <button
            aria-controls="long-menu"
            aria-haspopup="true"
            aria-label="More"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
